### PR TITLE
Surface site constraint

### PR DESCRIPTION
**Pull request by davidfarinajr**
### Motivation or Problem
The bidentate families can make molecules with a bunch of surface sites.  I was getting a bunch of edge species with 3 or 4 sites.  This PR adds a `maximumSurfaceSites` constraint if you want to limit the number of surface sites that can be in a species in your model.  

### Description of Changes
Added a `maximumSurfaceSites` constraint with default value `-1` (no constraint)

### Testing
Added a unit test and successfully used `maximumSurfaceSites = 2` when running RMG.

### CODE REVIEW

**Review by kblondal — DISMISSED** [review]
I like it, thanks David! I tested for a system I previously had many adsorbates with >2 surface sites in the edge. Including `maximumSurfaceSites=2` in the input file works as intended with these changes. The run time was a bit shorter while the core species and reactions stayed the same as without the constraint. As expected there are fewer edge species and reactions with the constraint. 

Based on that these changes made the model generation process a bit more efficient without compromising the final model. Currently thermo estimation for species with >2 surface sites isn't good and those will not be included in the core anyways. 

Before, without constraint:
```
Execution time (DD:HH:MM:SS): 00:00:03:37
Memory used: 815.93 MB
The final model core has 42 species and 202 reactions
The final model edge has 451 species and 1253 reactions
```

Now, with  `maximumSurfaceSites=2`
```
Execution time (DD:HH:MM:SS): 00:00:03:28
Memory used: 811.00 MB
The final model core has 42 species and 202 reactions
The final model edge has 392 species and 1156 reactions
```

**Review by rwest — APPROVED** [review]
Looks OK to me. David and Katrin have both used it successfully.

**Comment by davidfarinajr** [comment]
just added to documentation and rebased, hopefully the tests pass and we can merge it in 🤞

File: documentation/source/users/rmg/input.rst [hunk 1]
```diff
@@ -909,6 +909,7 @@ all of RMG's reaction families. ::
         maximumSiliconAtoms=2,
         maximumSulfurAtoms=2,
         maximumHeavyAtoms=10,
+		maximumSurfaceSites=2,
```
Comment (rwest) [inline documentation/source/users/rmg/input.rst:912]: tabs instead of spaces? renders strangely on github

Comment (davidfarinajr) [inline documentation/source/users/rmg/input.rst:912]: oops 😆

Comment (davidfarinajr) [reply documentation/source/users/rmg/input.rst]: yep, just realized that, should be fixed now

**Review by rwest — APPROVED** [review]
Looks fine to me

**Comment by rwest** [comment]
Merged this, but one other thought: maybe we should update our example input files to have a maximum of 2 ?

**Pull request merged**